cluster/mesos/docker/common/bin/await-file

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Wait for a file to exist in the filesystem.
+# Block up to the timeout duration in seconds (default: 10).
+# Usage: await-health-check [-t=<duration>] <address>
+# Requires: timeout, file
+
+set -e
+
+duration=10
+if [[ "$1" == "-t="* ]]; then
+    duration="${1:3}"
+    [ -z "$duration" ] && echo "Invalid duration supplied" && exit 1
+    shift
+fi
+
+file="$1"
+[ -z "$file" ] && echo "No file supplied" && exit 1
+
+# add the current dir to PATH
+bin=$(cd $(dirname $0) && pwd -P)
+export PATH=$PATH:${bin}
+
+echo "Waiting (up to ${duration}s) for ${file} to exist"
+if ! timeout ${duration} bash -c "while [ ! -f \"${file}\" ]; do sleep 0.5; done"; then
```

Comment:
That's actually the point. timeout will throw up because of a bad argument, rather than parsing it as part of the command to be executed.